src/Plugin/media/Source/Media.php

```diff
@@ -519,4 +535,278 @@ private function getMediaValue(MediaInterface $media, $attribute_name) {
     return $value;
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form = parent::buildConfigurationForm($form, $form_state);
+    /** @var \Drupal\media\MediaTypeInterface $entity */
+    $entity = $form_state->getFormObject()->getEntity();
+
+    $form['media_image_bundle'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Media image bundle'),
+      '#default_value' => $this->getConfiguration()['media_image_bundle'],
+      '#options' => $this->getImageMediaBundles(),
+      '#empty_option' => $this->t('- Select -'),
+      '#description' => $this->t('To have the thumbnail mapped to a media entity rather than a file (the default) choose a media type and field to map the thumbnail to.'),
+      '#ajax' => [
+        'callback' => [$this, 'mediaImageBundleOnChange'],
+        'event' => 'change',
+        'wrapper' => 'media-image-field',
+      ],
+    ];
+
+    if (!empty($this->getConfiguration()['media_image_bundle'])) {
+      $form['media_image_field'] = $this->getMediaImageFieldElement($this->getConfiguration()['media_image_bundle']);
+    }
+    else {
+      $form['placeholder_media_image_field'] = [
+        '#markup' => '<div id="media-image-field"></div>',
```

Comment:
Do we need to use `Html::UniqueId()` here?

Reply:
I don't think so. `Html::UniqueId()` will remember ids used across AJAX requests. We want an id that's consistent. 

I ended up having to play around with the AJAX stuff here a lot to get it to work. I ended up with the AJAX callback also having to return this placeholder div in the empty case. It's tricky b/c the AJAX callback doesn't seem to get the SubFormState, but the full form state, so it's unfortunately tightly coupled to the MediaType form :/.

I can prefix this id with the module name to mitigate somewhat.